test/blackbox-tests/test-cases/github11527.t/run.t

```diff
@@ -0,0 +1,17 @@
+Reproducing github #11527
+
+When running dune exec on a binary `bug` we can expect that the working directory and
+argv.(0) can be concatenated to get a valid path.
+
+When running dune exec from the root this is true.
+  $ dune exec -- bug
+  pwd: $TESTCASE_ROOT
+  exe: ./_build/install/default/bin/bug
+
+As expected, the `exe:` argument shows the correct path
+  $ (cd subdir && dune exec --root .. -- bug)
```

Comment:
I think it would be a bit nicer to just `mkdir subdir` here, then we don't have to have a `.gitkeep` dummy file to keep the path, but up to you.